File: tools/python-incident-analyzer/src/incident_analyzer/domain/investigation_note.py
"""Investigation note domain model."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class InvestigationNote:
    """Note added during an incident investigation."""

    author: str
    message: str
    created_at: str

    @classmethod
    def from_json(cls, data: dict[str, Any]) -> InvestigationNote:
        """Create an investigation note from backend JSON."""
        return cls(
            author=str(data.get("author", "")),
            message=str(data.get("message", "")),
            created_at=str(data.get("createdAt", "")),
        )

File: frontend/flutter-rules-practice/lib/features/incidents/domain/investigation_note.dart
class InvestigationNote {
  final String author;
  final String message;
  final String createdAt;

  const InvestigationNote({
    required this.author,
    required this.message,
    required this.createdAt,
  });

  factory InvestigationNote.fromJson(Map<String, dynamic> json) {
    return InvestigationNote(
      author: json['author'] as String? ?? '',
      message: json['message'] as String? ?? '',
      createdAt: json['createdAt'] as String? ?? '',
    );
  }
}